presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1149,6 +1170,7 @@ private ConnectorCommitHandle commitShared()
                         hdfsContext.getIdentity(),
                         hdfsContext.getQueryId().orElse(""),
                         hdfsContext.getClientInfo(),
+                        hdfsContext.getClientTags().orElse(new HashSet<>()),
```

Comment:
Collections.emptySet() instead of hashset if this doesn't need to be mutable

Reply:
Addressed all the immutable set concern, thanks